src/main/java/com/karankumar/bookproject/backend/entity/Book.java

```diff
@@ -101,7 +101,7 @@ public Book(String title, Author author, PredefinedShelf predefinedShelf) {
     }
     
     public void setEdition(Integer edition) {
-        if (edition == null) {
+        if (edition.isEmpty()) {
```

Comment:
Instead of that you can wrap the entire thing with in if condition checking edition != null